[Khaleelur Rahman] iP

Khaleelur-Rahman opened the pull request:
## ChatTasksMe
> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes)

ChatTasksMe frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😃 

And it is **ABSOLUTELY FREE**!

Features: 
 - [X] Managing tasks
 - [x] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the a sneak peek into the `main` method:
````
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Khaleelur-Rahman committed:
Add salutations.

Greet the person Hi and goodbye when the person runs the code or exits from it.

Khaleelur-Rahman committed:
Echo

Chatbot echos whatever the user inputs and leaves when the user inputs "Bye"

Khaleelur-Rahman committed:
Add, List

Lets user add todos and to display them.

Khaleelur-Rahman committed:
Mark as Done

Enabled users to mark tasksasd done and to undone tasks.
Renamed ToDoList class as Task class to encapsulate task info.

Khaleelur-Rahman committed:
Mark as done update

Updated the code to include return statements in task list instead of inside Duke class to make enforce abstraction.

Khaleelur-Rahman committed:
Todo, Deadline, Event tasks

Created 2 subclasses that inherit from task class to encapsulate the info related to each task in each class.

Khaleelur-Rahman committed:
Add I/O redirection technique.

Added automated testing to check if the the expected and actual output is correct thus far.

Khaleelur-Rahman committed:
Add automated text UI testing and error handling

Added some inputs to verify actual and expected output.
Handled errors for mark/unmark, list, deadline,event,todo commands.

Khaleelur-Rahman committed:
More exceptions

Handle mark and unmarked states.

Khaleelur-Rahman committed:
Add delete functionality

Support delete of tasks using arraylists. Added exceptions for the delete operation.

Khaleelur-Rahman committed:
Added enums

Created enums for all the 3 subtasks and edited toString in task class.

Khaleelur-Rahman committed:
Add enums

Added enums class for the 3 subtasks and edited toString accordingly.

Khaleelur-Rahman committed:
Add javadocs for all methods

Khaleelur-Rahman committed:
Adjusted Automated Text UI Testing

Khaleelur-Rahman committed:
Reduced code complexity

Reduced the number of functions.
Used startsWith() to ease code readability.
Added javadocs for modified methods.
Simplified main() method.

Khaleelur-Rahman committed:
Moved scanner object above while loop.

Khaleelur-Rahman committed:
Add save to harddisk feature

Used file write and read functions to save data to harddisk.

Khaleelur-Rahman committed:
Ignored Storage folder

Khaleelur-Rahman committed:
Delete src/main/storage directory

Khaleelur-Rahman committed:
Handle dates for deadline tasks

Khaleelur-Rahman committed:
Completed Level-8

Handled both date and time inputs for tasks.

Khaleelur-Rahman committed:
Revert "Completed Level-8"

This reverts commit 5981a0e4cb8d4aea5e985ec3b851c59cc7bd13b5.

Khaleelur-Rahman committed:
Revert "Revert "Completed Level-8""

This reverts commit f5ecb5361113574a10cdecfd122b57d8c9c00e02.

Khaleelur-Rahman committed:
Merge branch 'add-gradle-support' of https://github.com/Khaleelur-Rahman/ip

Khaleelur-Rahman committed:
Delete duke.txt

Khaleelur-Rahman committed:
Merge pull request #1 from Khaleelur-Rahman/branch-Level-8

Branch level 8

Khaleelur-Rahman committed:
Merge branch 'master' of https://github.com/Khaleelur-Rahman/ip

Khaleelur-Rahman committed:
Completed A-MoreOOPs level.

Khaleelur-Rahman committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Completed A-MoreOOPs level.

Khaleelur-Rahman committed:
merge

Khaleelur-Rahman committed:
Added test files for TaskList and Todo

Khaleelur-Rahman committed:
Added javaDocs

Khaleelur-Rahman committed:
Edited to comply with coding standard convention

Khaleelur-Rahman committed:
Merge branch 'A-CodingStandard'

Khaleelur-Rahman committed:
Completed Level-9

Khaleelur-Rahman committed:
Edited branch-Level-9

Khaleelur-Rahman committed:
Merge branch 'Branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java

Khaleelur-Rahman committed:
Update coding standards of code after adding stylechecks

WeeeHung commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,92 @@
+     * @throws IOException
+     */
+
+    public String handleCommand(String command, String input) throws DukeException, IOException {
```
The switch case and handling is very neat and well encapsulated :)

WeeeHung commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @Override
+    public String toString() {
```
perhaps formatted string will make this neater and easier to update changes in the future. ie. 
String formattedString = String.format("task: %s, start: %d, end: %.2f", task, start, end);

WeeeHung commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,47 @@
+     *
+     * @param e The exception containing the error message.
+     */
+    public void showLoadingError(Exception e) {
```
neat error handling!

WeeeHung commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,68 @@
+/**
+ * Class to handle the task
+ */
+public class Task {
```
i like that all task type classes are inherited from a parent class task, which prevents repeating many lines of code

WeeeHung commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,47 @@
+ * Class to take care of UI
+ */
+public class Ui {
+    private String divider = "------------------------------------\n";
```
could the following fields be FINAL, as they are pretty much constants that are not changes in the code

WeeeHung commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,36 @@
+public class TodoTest {
+
+    @Test
+    public void toDo_toString_success() {
```
great naming for error handling!

Goh-Li-Ting commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+    protected LocalTime endTime;
+
+    /**
+     * Constructor for Deadline class.
```
Perhaps there should be an empty line between description and parameter section.

Goh-Li-Ting commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+     * @param endTime end time of deadline task
+     * @param type enum type
+     */
+
```
Perhaps there should be no blank line between the documentation block and the method.

Goh-Li-Ting commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    /**
+     * Marks the task as isCompleted and returns a message indicating the task's new status.
```
Should it be 'Marks the task as **completed**' instead of 'isCompleted'?

Goh-Li-Ting commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+ * Class to maintain the deadline tasks entered by user.
+ */
+public class Deadline extends Task {
+    protected LocalDate deadline;
```
It was a little confusing differentiating between `deadline` and `endTime`. Perhaps you can consider renaming `deadline` to `endDate` instead so that it is more similar to `endTime`?

Goh-Li-Ting commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,481 @@
+    }
+
+    /**
+     * method to handle find operation
```
Perhaps it would be better to start the method description using a verb.

Goh-Li-Ting commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,481 @@
+    /**
+     * method to handle find operation
+     * @param input user's input
+     * @return string representation of operayion
```
There seems to be a typo 'operayion' here

Goh-Li-Ting commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,65 @@
+ * Class to handle event task
+ */
+public class Event extends Task {
+    protected LocalDate start;
```
Similar to `Deadline`, perhaps you can rename `start` as `startDate` and `end` as `endDate`

Khaleelur-Rahman committed:
Complete Level-10.

Khaleelur-Rahman committed:
Edit code to match coding standards

Khaleelur-Rahman committed:
Add Ui class to handle responses

Khaleelur-Rahman committed:
Edit Ui class

Khaleelur-Rahman committed:
Edit some javaFx to make the dialog box look nicer

Khaleelur-Rahman committed:
Add assertions to TaskList class.

Assertions were added to check if the return toString()
is empty.

If yes, they will cause issues in the method flow. Hence
asserted.

This way, the progrom flow is ensured without any major bugs.

Khaleelur-Rahman committed:
Merge pull request #2 from Khaleelur-Rahman/A-Assertions

Add assertions to TaskList class.

Khaleelur-Rahman committed:
Refactor code

Simplified code to ease understanding and to reduce
number of LOC in each method.

Separated the handle methods into two to handle
both user and file actions.

Khaleelur-Rahman committed:
Merge branch 'master' into A-CodeQuality

Khaleelur-Rahman committed:
Merge pull request #3 from Khaleelur-Rahman/A-CodeQuality

Refactor code

Khaleelur-Rahman committed:
Add better search features

Users will want to have better find features to
not just search for the exact keyword, but also
similar words.

Khaleelur-Rahman committed:
Edit boolean array to ensure coding standard

Khaleelur-Rahman committed:
Remove unwanted imports

Khaleelur-Rahman committed:
Edit GUI

Differentiated user input and duke response with
different background colour.

Khaleelur-Rahman committed:
Adjust dialogBox width and other minor changes

Gave the dialogBox a fixed width and edited the event and deadline
classes to store day in 'dd' format instead of 'd' format

Khaleelur-Rahman committed:
Made adjustments to the GUI

Khaleelur-Rahman committed:
Merge branch 'branch-A-BetterGui'

Khaleelur-Rahman committed:
Add personality to Duke

Added a ronaldo personality to say 'SUI' before each response.

Khaleelur-Rahman committed:
Add errors to handle

Errors such as command format and data not in required format added.

Khaleelur-Rahman committed:
Add test cases for Event and Deadline classes

Added more test cases and javadocs for parser class.

Khaleelur-Rahman committed:
Create sub-packages inside duke package

Khaleelur-Rahman committed:
Fix newline UI issues

Khaleelur-Rahman committed:
Add additional classes to reduce code complexity

Added SearchInTasks class to handle the find extension.
Added DateAndTimeHandler class to handle the parsing and checking
of date and time.

Reduce the complexity of Duke.java by shifting the storage methods to
storage class.

Khaleelur-Rahman committed:
Add test cases and stylechecks

Khaleelur-Rahman committed:
Fix bugs in storage.java

Khaleelur-Rahman committed:
Add screenshot of Duke

Khaleelur-Rahman committed:
Change name of product to be RonaldoSaysDo

Khaleelur-Rahman committed:
Create storage file if it does not exsit

Khaleelur-Rahman committed:
Fix bugs related to unmark command

Khaleelur-Rahman committed:
Initial README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Create Jar File

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Update README.md

Khaleelur-Rahman committed:
Edit package name and create new jar file

Khaleelur-Rahman committed:
Gradle changes for release

Khaleelur-Rahman committed:
Update product name in README.md

Khaleelur-Rahman committed:
Add features and usages section in README.md

Khaleelur-Rahman committed:
Typo in README.md

Khaleelur-Rahman committed:
Final changes

Reduce the size of handler methods to be within 30 LoC

Khaleelur-Rahman committed:
Merge branch 'master' of https://github.com/Khaleelur-Rahman/ip
